### PR TITLE
binary encoding

### PR DESCRIPTION
**Pull request by jelyoussefi**

### CODE REVIEW

**Comment by wagmarcel** [comment]
@jelyoussefi please squash to only one commit (if there is no specific reason why you made 2 commits) and add more detailed comments what you changed.
In addition, please signoff the commit.

File: api/rest/iot.connection.def.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ module.exports = function(config) {
      * Top of the hierarchy. Common attributes to every
      * Connection Options
      */
-    function ConnectionOptions() {
+    function ConnectionOptions(isBinary) {
```
Comment (srware) [inline api/rest/iot.connection.def.js]: I don't think the "isBinary" flag needs to be sent to this function as you are overriding the body and headers whenever this is used anyway so what gets set here is irrelevant...

File: api/rest/data.def.js [hunk 2]
```diff
@@ -32,11 +34,17 @@ module.exports = function(config) {
...
 
     function SubmitDataOption(data) {
+        var isBinary = common.isBinary(data.body);
         this.pathname = common.buildPath(api.data.SEND, data.deviceId);
         this.token = data.userToken;
-        ConnectionOptions.call(this);
+        ConnectionOptions.call(this, isBinary);
         this.method = 'POST';
-        this.body =  JSON.stringify(data.body);
+        if ( isBinary ) {
```
Comment (srware) [inline api/rest/data.def.js]: Make sure you run "grunt" before creating a PR to test for code consistency, formatting and ensure all code tests pass. This if..else.. doesn't conform to the formatting for the project.

File: package.json [hunk 3]
```diff
@@ -1,29 +1,55 @@
 {
```
Comment (srware) [inline package.json]: The only modifications to the package.json I am likely to accept in a PR is additional dependencies required for the code being contributed.

File: package.json [hunk 4]
```diff
@@ -1,29 +1,55 @@
...
     "grunt-contrib-compress": "^1.4.3",
-    "gruntify-eslint": "^4.0.0",
     "grunt-license-finder": "^2.0.0",
     "grunt-mocha-istanbul": "^5.0.2",
+    "gruntify-eslint": "^4.0.0",
     "istanbul": "^0.4.5",
     "mocha": "^4.0.1",
-    "rewire": "^2.5.2"
+    "rewire": "^2.5.2",
+    "cbor": "^4.1.5"
```
Comment (srware) [inline package.json]: Everything but this line needs to be removed from the PR.

**Pull request closed without merging**